### PR TITLE
feat(deployment): deploy `SignalService` in L1 / L2

### PR DESCRIPTION
**Pull request by davidtaikocha**
plz ignore the unit test errors at this point. `pnpm deploy:hardhat` and `pnpm test:genesis` are all passed locally.

### CODE REVIEW

File: packages/protocol/tasks/deploy_L1.ts [hunk 1]
```diff
@@ -184,6 +184,18 @@ export async function deployContracts(hre: any) {
...
 
+    // SignalService
+    const SignalService = await deploySignalSerive(hre, AddressManager.address);
+
+    // Used by Bridge
+    await utils.waitTx(
+        hre,
+        await AddressManager.setAddress(
+            `${chainId}.signal_service`,
+            Bridge.address
```
Comment (dantaik) [inline packages/protocol/tasks/deploy_L1.ts:195]: `Bridge.address` is not correct, we should deploy a SignalService then use its address.

File: packages/protocol/tasks/deploy_L1.ts [hunk 2]
```diff
@@ -258,3 +266,18 @@ async function deployTokenVault(
...
+
+async function deploySignalSerive(
+    hre: any,
+    addressManager: string
+): Promise<any> {
+    const libTrieProof = await utils.deployContract(hre, "LibTrieProof");
+
+    const SignalService = await utils.deployContract(hre, "SignalService", {
+        LibTrieProof: libTrieProof.address,
+    });
```
Comment (dantaik) [inline packages/protocol/tasks/deploy_L1.ts:278]: call `init`?

File: packages/protocol/tasks/deploy_L1.ts (hunk 1 above)
Comment (davidtaikocha) [reply packages/protocol/tasks/deploy_L1.ts]: fixed

File: packages/protocol/tasks/deploy_L1.ts (hunk 2 above)
Comment (davidtaikocha) [reply packages/protocol/tasks/deploy_L1.ts:278]: `init` is called in next line.

**Review by dantaik — APPROVED** [review]

**Pull request merged**